docs/getting-started-guides/coreos/azure/lib/azure_wrapper.js

```diff
@@ -107,6 +107,9 @@ var create_ssh_key = function (prefix) {
   };
   openssl.exec('req', opts, function (err, buffer) {
     if (err) console.log(clr.red(err));
+    openssl.exec('rsa', { in: opts.keyout, out: opts.keyout }, function (err, buffer) {
+       if (err) console.log(clr.red(err));
+    });
     fs.chmod(opts.keyout, '0600', function (err) {
```

Comment:
Isn't it possible that the chmod and 'openssl rsa' happen in parallel and the 'openssl rsa' command clobbers the permission change, leaving the file world readable?